test/lit/passes/merge-blocks-eh.wast

```diff
@@ -0,0 +1,214 @@
+  ;; CHECK-NEXT:    (call $import)
+  ;; CHECK-NEXT:    (unreachable)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $drop-block-try_catch_all_ref
+    ;; This block is dropped, so the try_table's exnref value can be removed
+    ;; by replacing catch_all_ref with catch_all.
+    (drop
+      (block $catch (result exnref)
+        (try_table (catch_all_ref $catch)
+          (call $import)
+          (unreachable)
+        )
+      )
+    )
+  )
+
+  ;; CHECK:      (func $drop-block-try_catch_ref (type $0)
+  ;; CHECK-NEXT:  (block $catch
+  ;; CHECK-NEXT:   (try_table (catch $empty $catch)
+  ;; CHECK-NEXT:    (call $import)
+  ;; CHECK-NEXT:    (unreachable)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $drop-block-try_catch_ref
+    ;; As above, but with catch_ref instead of catch_all_ref. We can still
+    ;; optimize.
+    (drop
+      (block $catch (result exnref)
+        (try_table (catch_ref $empty $catch)
+          (call $import)
+          (unreachable)
+        )
+      )
+    )
+  )
+
+  ;; CHECK:      (func $drop-block-try_catch_multi (type $0)
+  ;; CHECK-NEXT:  (block $catch
+  ;; CHECK-NEXT:   (try_table (catch $empty $catch) (catch_all $catch)
+  ;; CHECK-NEXT:    (call $import)
+  ;; CHECK-NEXT:    (unreachable)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $drop-block-try_catch_multi
+    ;; As above, but with two catches, both of whom can be optimized.
+    (drop
+      (block $catch (result exnref)
+        (try_table (catch_ref $empty $catch) (catch_all_ref $catch)
+          (call $import)
+          (unreachable)
+        )
+      )
+    )
+  )
+
+  ;; CHECK:      (func $drop-block-try_catch_all_i32 (type $0)
+  ;; CHECK-NEXT:  (tuple.drop 2
+  ;; CHECK-NEXT:   (block $catch (type $1) (result i32 exnref)
+  ;; CHECK-NEXT:    (try_table (catch_ref $i32 $catch)
+  ;; CHECK-NEXT:     (call $import)
+  ;; CHECK-NEXT:     (unreachable)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $drop-block-try_catch_all_i32
+    ;; Send a tag value + exnref. We don't handle this yet TODO
+    (tuple.drop 2
+      (block $catch (result i32 exnref)
+        (try_table (catch_ref $i32 $catch)
+          (call $import)
+          (unreachable)
+        )
+      )
+    )
+  )
+
+  ;; CHECK:      (func $drop-block-try_catch_multi_partial (type $0)
+  ;; CHECK-NEXT:  (tuple.drop 2
+  ;; CHECK-NEXT:   (block $outer (type $1) (result i32 exnref)
+  ;; CHECK-NEXT:    (block $inner
+  ;; CHECK-NEXT:     (try_table (catch_ref $i32 $outer) (catch_all $inner)
+  ;; CHECK-NEXT:      (call $import)
+  ;; CHECK-NEXT:      (unreachable)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (unreachable)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $drop-block-try_catch_multi_partial
+    ;; Two possible tags can be thrown and caught, and we can optimize one.
```

Comment:
One does not have a tag (because it's `catch_all_ref`)

Reply:
Thanks, fixed.